src/cargo/util/toml_mut/manifest.rs

```diff
@@ -540,5 +540,42 @@ fn non_existent_dependency_err(
 }
 
 fn remove_array_index(array: &mut toml_edit::Array, index: usize) {
-    array.remove(index);
+    let value = array.remove(index);
+
+    // Captures all lines before leading whitespace
+    let prefix_lines = value
+        .decor()
+        .prefix()
+        .and_then(|p| p.as_str().expect("spans removed").rsplit_once('\n'))
+        .map(|(lines, _current)| lines);
+    // Captures all lines after trailing whitespace, before the next comma
+    let suffix_lines = value
+        .decor()
+        .suffix()
+        .and_then(|p| p.as_str().expect("spans removed").split_once('\n'))
+        .map(|(_current, lines)| lines);
+    let mut merged_lines = String::new();
+    if let Some(prefix_lines) = prefix_lines {
+        merged_lines.push_str(prefix_lines);
+        merged_lines.push('\n');
+    }
+    if let Some(suffix_lines) = suffix_lines {
+        merged_lines.push_str(suffix_lines);
+        merged_lines.push('\n');
+    }
+
+    let next_index = index; // Since `index` was removed, that effectively auto-advances us
+    if let Some(next) = array.get_mut(next_index) {
+        let next_decor = next.decor_mut();
+        let next_prefix = next_decor
+            .prefix()
+            .map(|s| s.as_str().expect("spans removed"))
```

Comment:
question for `toml_edit`. Since `RawString` is opaque to users, what does "span removed" mean to them (or me, I don't quite follow)?

Reply:
When `toml_edit` does its parsing pass, `RawString`s are just spans (`Range<usize>`).  This allows `toml` to never allocate for decor as it just gets abandoned. We then replace all of these spans with the actual content when generating a `toml_edit::Document` so you can copy a `Value` out of one document and insert it into another without getting bounds issues.  Since its unused, we save on memory by tracking this in an enum. 

We wouldn't want to change out the internal string type when switching away from spans because that would require re-allocating everything around the string (the array, table, etc).

We wouldn't want to have a separate public/private API that makes these assumptions for the user because then we need to make sure the public isn't used internally.

So this gets exposed to users.  In short, `RawString` is too low level to know of the details of when its safe but the caller easily does.